posts/2024-05-06-jakarta-nosql-in-action-meet-jnopo-game.adoc

```diff
@@ -0,0 +1,828 @@
+
+[source, java]
+----
+package org.jakartaee.sample.model;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import org.jakartaee.sample.game.GameOver;
+
+@ApplicationScoped
+public class GameMatchCapturer {
+
+    public void captureAndPersist(@Observes GameOver gameOver){
+        // put the persistence logic here...
+    }
+
+}
+
+----
+
+At this point, let's instantiate a new `GameMatch` entity instance, populate it from `GameOver` data and then store it into the database by using the `Playoffs` component.
+
+The required `Playoff` component, which is a repository implementation offered by Jakarta Data specification implementation, Eclipse JNoSQL, will be injected by CDI using the `@Inject` and `@Database(DOCUMENT)` annotations:
+
+[source, java]
+----
+package org.jakartaee.sample.model;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import org.eclipse.jnosql.mapping.Database;
+import org.eclipse.jnosql.mapping.DatabaseType;
+import org.jakartaee.sample.game.GameOver;
+
+@ApplicationScoped
+public class GameMatchCapturer {
+
+    @Inject
+    @Database(DatabaseType.DOCUMENT)
+    Playoffs playoffs;
+
+    public void captureAndPersist(@Observes GameOver gameOver){
+
+            var gameMatch  = new GameMatch(
+                    gameOver.gameId(),
+                    PlayerInfo.of(gameOver.playerAInfo()),
+                    PlayerInfo.of(gameOver.playerBInfo()),
+                    gameOver.winnerInfo().map(PlayerInfo::of).orElse(PlayerInfo.NOBODY),
+                    gameOver.winnerInfo().map(PlayerInfo::of).orElse(PlayerInfo.NOBODY),
+                    gameOver.isTied()
+            );
+
+            playoffs.add(gameMatch);
+    }
+
+}
+----
+
+Now, let's get started the  https://openliberty.io/[Open Liberty] runtime to figure out if the game match results are being stored on the MongoDB:
+
+* On Linux/Mac machines, perform the following command:
++
+[source, bash]
+----
+./mvnw liberty:dev
+----
+
+* On Windows machines, perform the following command:
++
+[source, powershell]
+----
+mvnw.cmd liberty:dev
+----
+
+After some game matches, we can confirm the persistence of the game matches into the database by looking at Mongo Express running on http://localhost:8081 :
+
+[.img_border_light]
+image::/img/blog/mongo-express-01.png[Mongo Express 01, align="center" width=80%]
+
+[.img_border_light]
+image::/img/blog/mongo-express-02.png[Mongo Express 02, align="center" width=80%]
+
+[.img_border_light]
+image::/img/blog/mongo-express-03.png[Mongo Express 03, align="center" width=80%]
+
+[.img_border_light]
+image::/img/blog/mongo-express-04.png[Mongo Express 04, align="center" width=80%]
+
+Or, if you're a command-line practitioner, you can check the data by using the `mongosh` cli:
```

Comment:
Unsure if we need both, as it may confuse people thinking they need to do both the UI and command line. Maybe best to just stick with UI for now? @lauracowen WDYT?